spec.md

```diff
@@ -1260,6 +1276,7 @@ The following parameters SHOULD be specified on the request:
 | `name` | path  | Name of the target repository.                                                              |
 | `n`    | query | Limit the number of entries in each response. It not present, all entries will be returned. |
 | `last` | query | Result set will include values lexically after last.                                        |
+| `ns`   | query | (OPTIONAL) Namespace of repository. SHOULD be set to mirrored host.                         |
```

Comment:
so this SHOULD be `docker.io/` or `quay.io/`

Reply:
`docker.io` or `quay.io`, the `/` as a separator is implied by use of paths in the repository name. But can make that explicit.

Reply:
This project is new to me, but I wanted to share some information as this naming convention was a bit confusing to me.
In Quay, `namespace` references a particular User or Organization. For example, given the URI `quay.io/redhat/quay`
- `quay.io` is the hostname/registry
- `redhat` is the namespace
- `quay` is the repository

Reply:
Yes, the word "namespace" is quite unspecific and its usage here partially conflicts with other appearances of that word within the spec. Particularly, under "Starting an upload", the request `POST /v2/<name>/blobs/uploads/` has a description reading:

> The parameters of this request are the image **namespace** under which the layer will be linked.

That's a different kind of namespace than what is meant here (and even different from how Quay uses the term apparently). This passes into #27 territory and should be cleaned up anyway, but IMO we shouldn't make it worse and use such vague terms in more places.

If the `ns` parameter only has values like `docker.io` or `quay.io`, would it make sense to rename it e.g. `upstream` and explicitly define it as containing the _domain name_ of the upstream registry?

Reply:
+1 or, alternatively, `upstream-registry` to be specific

Reply:
> Yes, the word "namespace" is quite unspecific and its usage here partially conflicts with other appearances of that word within the spec. Particularly, under "Starting an upload", the request POST /v2/<name>/blobs/uploads/ has a description reading:

The word "namespace" is intentionally unspecific since it is highly contextual. As pointed out elsewhere, we specify a the `<name>` component in the URL path, the `ns` parameter is further information related to that `<name>`, hence the full name being `<namespace>/<name>`. We should clarify any part of the documentation that suggests that full names are `<host>/<namespace>/<repository>` as that is only referring to a single registry interpretation and this specification is intended to be less opinionated.

>  would it make sense to rename it e.g. upstream and explicitly define it as containing the domain name of the upstream registry

It would be better to clarify how it is applied rather than rename it. The namespace could be the upstream registry, and should be used by mirrors to resolve the host. However, just because this normally represents a domain, does not mean it represents the upstream host. For example, with `docker.io` the upstream host is resolved differently. The intent of this change is to provide the full name used by the client, not to tell the registry host how to use the name. Mirroring is just the use case here, the specification itself is more generic.